src/domain/common/whiteboard/excalidraw/CollaborativeExcalidrawWrapper.tsx

```diff
@@ -121,24 +131,85 @@ const CollaborativeExcalidrawWrapper = ({
 
   const { user } = useUserContext();
-  const username = user?.user.profile.displayName ?? 'User';
+  const username = useMemo(() => user?.user.profile.displayName ?? 'User', [user?.user.profile.displayName]);
 
   const [isSceneInitialized, setSceneInitialized] = useState(false);
 
-  const handleScroll = useRef(
-    debounce(async () => {
-      excalidrawApi?.refresh();
-    }, WINDOW_SCROLL_HANDLER_DEBOUNCE_INTERVAL)
-  ).current;
+  const [excalidrawApi, setExcalidrawApi] = useState<ExcalidrawImperativeAPI | null>(null);
+
+  // @@@ WIP ~ #7611
+  // ------------------------------------------------------------------------------------------------------------------
+  /**
+   * 	Key Observations and Adjustments:
+   *
+   * 	excalidrawApi Availability: We are correctly using useState to manage the excalidrawApi, and it's initialized asynchronously within the handleInitializeApi callback. This means the excalidrawApi might not be immediately available when the component mounts.
+   *
+   * 	handleScroll Purpose: Our handleScroll function calls excalidrawApi?.refresh(). It's important to understand why we're calling refresh(). Based on the Excalidraw documentation, refresh() is typically used to re-render the Excalidraw component when its container size changes or when something else outside of Excalidraw's internal state affects its rendering. If we're calling it solely on window scroll, we should carefully consider whether it's actually necessary for our use case. Frequent calls to refresh() can impact performance.
+   *
+   * 	Collab API Integration: The collabApi and its onPointerUpdate function are passed to Excalidraw. This is good.
+   *
+   * 	whiteboard Key: We are correctly using the whiteboard.id as the key to the <Excalidraw> component, forcing a re-render when the whiteboard changes.
+   *
+   * 	Dependency Array Updates: We now pass excalidrawApi to the debounced function. So excalidrawApi becomes important dependency in the useEffect.
+   */
+
+  const debouncedRefresh = useRef<DebouncedFunc<() => void> | null>(null) as MutableRefObject<DebouncedFunc<
+    () => void
+  > | null>; // Explicitly cast to MutableRefObject
 
   useEffect(() => {
-    window.addEventListener('scroll', handleScroll, true);
+    debouncedRefresh.current = debounce(() => {
+      if (excalidrawApi) {
+        excalidrawApi.refresh();
+      } else {
+        console.warn('excalidrawApi is not available.');
+      }
+    }, WINDOW_SCROLL_HANDLER_DEBOUNCE_INTERVAL);
+
+    const handleScroll = () => {
+      debouncedRefresh.current?.();
+    };
+
+    window.addEventListener('scroll', handleScroll, { passive: true, capture: true });
 
     return () => {
-      handleScroll.cancel();
-      window.removeEventListener('scroll', handleScroll, true);
+      debouncedRefresh.current?.cancel();
+      window.removeEventListener('scroll', handleScroll, { capture: true });
     };
-  }, [handleScroll]);
+  }, [excalidrawApi]);
```

Comment:
Depending on `excalidrawApi` will run `useEffect` twice.
Depending on `handleScroll` will run it once.

`excalidrawApi?.refresh();` will be transpiled to `(_a = excalidrawApi) === null || _a === void 0 ? void 0 : _a.refresh();` which is the same as using and if statement with neglectable performance benefit for the if statement, which you can do in the function definition also by the way.
Maybe profiling both approaches would shed some light.

IMO I wouldn't consider this change as a safer or faster option.